src/net.md

```diff
@@ -876,6 +877,85 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-extract-mediawiki-links]: #ex-extract-mediawiki-links
+<a name="ex-extract-mediawiki-links"/>
+## Extract all unique links from a MediaWiki markup
+
+[![reqwest-badge]][reqwest] [![regex-badge]][regex] [![cat-net-badge]][cat-net]
+
+Pull the source of a Mediawiki page using [`reqwest::get`] and then
+find all entries of internal and external links in that source with
+[`Regex::captures_iter`].
+
+In Mediawiki syntax:
+* Internal links take a form of `[[Page]]` or `[[Page|link text]]`.
+* External links are often `[https:// Link text]`, `|url=https://`
+or sometimes `{{URL|https://}}`.
+
+Internal links are case insensitive, but URLs are not, and this
+affects how we should deduplicate them.
+
+```rust,no_run
+# #[macro_use]
+# extern crate error_chain;
+extern crate reqwest;
+extern crate regex;
+
+use std::io::Read;
+use regex::Regex;
+
+# error_chain! {
+#     foreign_links {
+#         Io(std::io::Error);
+#         Reqwest(reqwest::Error);
+#         Regex(regex::Error);
+#     }
+# }
+# 
+fn run() -> Result<()> {
+    let mut resp = reqwest::get(
+        "https://en.wikipedia.org/w/index.php?title=Rust_(programming_language)&action=raw",
+    )?;
+
+    let mut content = String::new();
+    resp.read_to_string(&mut content)?;
+
+    let mut links = Vec::new();
+    let mut urls = Vec::new();
+
+    // Capture the part of [[Page|link text]] that is inside the brackets
+    for mat in Regex::new(r#"\[\[(?P<link>.+?\|?)\]\]"#)?
```

Comment:
That was the plan but I've tried <pre>\[\[(.+?)(\|?.*?)\]\]</pre> and the first group is too lazy and captures only the first character. I'll research further I guess.